qa/zcash/smoke_tests.py

```diff
@@ -0,0 +1,800 @@
+
+        self.__datadir = datadir
+        self.__wallet = wallet
+        self.__zcashd = zcashd
+        self.__zcash_cli = zcash_cli
+        self.__process = None
+        self.__proxy = None
+
+    def start(self, testnet=True, extra_args=None, timewait=None):
+        if self.__proxy is not None:
+            raise RuntimeError('Already started')
+
+        rpcuser = 'st'
+        rpcpassword = 'st'
+
+        args = [
+            self.__zcashd,
+            '-datadir=%s' % self.__datadir,
+            '-wallet=%s' % self.__wallet,
+            '-rpcuser=%s' % rpcuser,
+            '-rpcpassword=%s' % rpcpassword,
+            '-showmetrics=0',
+            '-experimentalfeatures',
+            '-zmergetoaddress',
+        ]
+        if testnet:
+            args.append('-testnet=1')
+        if extra_args is not None:
+            args.extend(extra_args)
+
+        self.__process = subprocess.Popen(args)
+
+        cli_args = [
+            self.__zcash_cli,
+            '-datadir=%s' % self.__datadir,
+            '-rpcuser=%s' % rpcuser,
+            '-rpcpassword=%s' % rpcpassword,
+            '-rpcwait',
+        ]
+        if testnet:
+            cli_args.append('-testnet=1')
+        cli_args.append('getblockcount')
+
+        devnull = open('/dev/null', 'w+')
+        if os.getenv('PYTHON_DEBUG', ''):
+            print('start_node: zcashd started, calling zcash-cli -rpcwait getblockcount')
+        subprocess.check_call(cli_args, stdout=devnull)
+        if os.getenv('PYTHON_DEBUG', ''):
+            print('start_node: calling zcash-cli -rpcwait getblockcount returned')
+        devnull.close()
+
+        rpcuserpass = '%s:%s' % (rpcuser, rpcpassword)
+        rpchost = '127.0.0.1'
+        rpcport = 18232 if testnet else 8232
+
+        url = 'http://%s@%s:%d' % (rpcuserpass, rpchost, rpcport)
+        if timewait is not None:
+            self.__proxy = Proxy(url, timeout=timewait)
+        else:
+            self.__proxy = Proxy(url)
+
+    def stop(self):
+        if self.__proxy is None:
+            raise RuntimeError('Not running')
+
+        self.__proxy.stop()
+        self.__process.wait()
+        self.__proxy = None
+        self.__process = None
+
+    def __getattr__(self, name):
+        if self.__proxy is None:
+            raise RuntimeError('Not running')
+
+        return self.__proxy.__getattr__(name)
+
+
+#
+# Test driver
+#
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--list-stages', dest='list', action='store_true')
+    parser.add_argument('--mainnet', action='store_true', help='Use mainnet instead of testnet')
+    parser.add_argument('--wallet', default='wallet.dat', help='Wallet file to use (within data directory)')
+    parser.add_argument('datadir', help='Data directory to use for smoke testing', default=None)
+    parser.add_argument('stage', nargs='*', default=STAGES,
+                        help='One of %s'%STAGES)
```

Comment:
Nit: Missing whitespace around `%`. Also, there is no need to line-wrap here (without wrapping this line is still shorter than the three before it).